thorlcr/graph/thgraphmaster.cpp

```diff
@@ -2722,6 +2723,19 @@ void CMasterGraph::handleSlaveDone(unsigned node, MemoryBuffer &mb)
         sdMb.setBuffer(len, (void *)d);
         act->slaveDone(node, sdMb);
     }
+    Owned<IThorActivityIterator> iter = getConnectedIterator();
+    ForEach(*iter)
+    {
+            CGraphElementBase &element = iter->query();
+            CActivityBase *activity = element.queryActivity();
+            if (activity)
+            {
+                stat_type nodeSpillSize = activity->getNodeSpillSize(node);
+                nodeActivityTotalSpillSize += nodeSpillSize;
+                if (nodeSpillSize>nodeActivityPeakSpillSize)
```

Comment:
Peak will need to be calculated differently.
Spilling is dynamic, it will grow and shrink in the lifetime of a subgraph, e.g. hash dedup will has spill phases and spill bucket that get dumped and consumed, sort has spill phases, then merges those spill phases, and deletes the earlier phase.
So the total ever spilt activity, is not a high water mark.

NB: This is less true of inter-graph spilling, but also could be the case there, e.g. if there are multiple arms reading spills, then some may spill and consume (delete) upstream spills from other subgraphs, in other words, the peak can't be seen at the end of the subgraph.